backend/__test__/helper.test.ts

```diff
@@ -0,0 +1,210 @@
+import { JsonObject } from "swagger-ui-express";
+import { expect, describe, it } from "@jest/globals";
+
+describe("CompareSingleJson Helper function works", () => {
+  it("Check all queries", async () => {
+    // CompareSingleJson tests
+    expect(compareSingleJson(query_type1, query_type2)).toBe(false);
+    expect(compareSingleJson(query_type1, query2_type1)).toBe(true);
+    expect(compareSingleJson(query_type1, query_type2)).toBe(false);
+    expect(compareSingleJson(query_type1, query_type3)).toBe(false);
+    expect(compareSingleJson(query_type1, query_type4)).toBe(false);
+    expect(compareSingleJson(query_type1, query_type1)).toBe(true);
+  });
+});
+
+describe("checkArrayContainsObjects Helper function works", () => {
+  it("Check all arrays", async () => {
+    // checkArrayContainsObjects tests
+    expect(
+      checkArrayContainsObjects(
+        [query_type1, query_type2],
+        [query_type1, query_type2]
+      )
+    ).toBe(true);
+    expect(
+      checkArrayContainsObjects(
+        [query_type1, query_type2],
+        [query_type2, query_type1]
+      )
+    ).toBe(true);
+    expect(
+      checkArrayContainsObjects(
+        [query2_type1, query_type1],
+        [query_type1, query_type2]
+      )
+    ).toBe(true);
+    expect(
+      checkArrayContainsObjects(
+        [query2_type1, query_type1],
+        [query_type3, query_type2]
+      )
+    ).toBe(false);
+    expect(
+      checkArrayContainsObjects(
+        [query_type3, query_type2, query_type1],
+        [query_type2, query_type1]
+      )
+    ).toBe(false);
+    expect(
+      checkArrayContainsObjects(
+        [query_type3, query_type2, query_type1],
+        [query_type3, query_type1, query_type2]
+      )
+    ).toBe(true);
+  });
+});
+
+// Written with the help of ChatGPT
+export function compareSingleJson(
+  obj1: JsonObject,
+  obj2: JsonObject,
+  enableLogging = false
+): boolean {
+  // Check if objects are of the same type
+  if (typeof obj1 !== typeof obj2) {
```

Comment:
for what are you checking for just for? just whether each value has the same type?